storages/jsonl-storage/src/store_mut.rs

```diff
@@ -168,3 +186,59 @@ where
         .map(|item| Ok(item?.1))
     }
 }
+
+impl JsonlStorage {
+    async fn append_data_json(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist)?;
+
+        let json_path = self.json_path(table_name);
+        let column_defs = schema.column_defs.unwrap_or_default();
+        let labels = column_defs
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(json_path)
+            .map_storage_err()?;
+
+        let json_array = JsonValue::Array(
+            rows.into_iter()
+                .map(|row| match row {
+                    DataRow::Vec(values) => labels
+                        .iter()
+                        .zip(values.into_iter())
+                        .map(|(key, value)| Ok((key.to_string(), value.try_into()?)))
+                        .collect::<Result<Map<String, JsonValue>>>(),
+                    DataRow::Map(hash_map) => hash_map
+                        .into_iter()
+                        .map(|(key, value)| Ok((key, value.try_into()?)))
+                        .collect(),
+                })
+                .map(|result| result.map(JsonValue::Object))
+                .collect::<Result<Vec<_>>>()?,
+        );
+        let json_array = to_string_pretty(&json_array).map_storage_err()?;
+
+        writeln!(file, "{json_array}").map_storage_err()?;
```

Comment:
Rather than using `writeln!`, how about `file.write_all`?
to use `write!` or `writeln!`, we need to do unnecessary string to string print but we don't need that if we use `write_all`
```rust
file.write_all(json_array.as_bytes()).map_storage_err()?;
```